tool/tctl/common/auth_command.go

```diff
@@ -926,14 +925,32 @@ func (a *AuthCommand) checkProxyAddr(ctx context.Context, clusterAPI auth.Client
 			return nil
 		}
 
-		uaddr, err := utils.ParseAddr(addr)
+		_, err := utils.ParseAddr(addr)
 		if err != nil {
 			log.Warningf("Invalid public address on the proxy %q: %q: %v.", p.GetName(), addr, err)
 			continue
 		}
+
+		ping, err := webclient.Ping(
+			&webclient.Config{
+				Context:   ctx,
+				ProxyAddr: addr,
+				Insecure:  true,
```

Comment:
why insecure?

Reply:
We are dialing to the proxy public address and it might use a self-signed certificate.

`tctl` has an insecure flag but it's supposed to control the auth server certificate, not the proxy.

Reply:
But this is actually insecure right? I don't understand why it would be okay to go insecure by default here

Reply:
We use the proxies registered in auth server, so at least they have valid credentials to connect to our cluster but I dropped the insecure.
If the connection fails, users are forced to provide the `--proxy` flag.

Reply:
thanks for the update, yeah we can be sure that's the real proxy address but we still want to be protected from MITM attacks